From 82066e5f46d743e7a7cf396be7f95e577084b93e Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@kernel.org>
Date: Thu, 6 Nov 2025 08:10:27 +0100
Subject: [PATCH] build: avoid breaking cross builds with OpenSSL enabled

The OpenSSL sanity check wants to be executed, thus we should only try
this if it's possible. Since this is just a warning, we can just skip it
if it is not possible.

Signed-off-by: Daniel Wagner <wagi@kernel.org>

Upstream: https://github.com/linux-nvme/libnvme/pull/1090

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index dce5c68e..6d1f5204 100644
--- a/meson.build
+++ b/meson.build
@@ -252,7 +252,7 @@ else
   conf.set('fallthrough', 'do {} while (0) /* fallthrough */')
 endif
 
-if openssl_dep.found()
+if openssl_dep.found() and meson.can_run_host_binaries()
   if openssl_dep.type_name() != 'internal'
     # Check for a bug in the EVP_PKEY_CTX_add1_hkdf_info implementation
     res = cc.run(
